tests/ecmp/test_fgnhg.py

```diff
@@ -141,168 +139,514 @@ def create_fg_ptf_config(ptfhost, ip_to_port, port_list, bank_0_port, bank_1_por
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.add_nh.log".format(test_time)
 
     ptf_runner(ptfhost,
             "ptftests",
             "fg_ecmp_test.FgEcmpTest",
             platform_dir="ptftests",
             params={"test_case": 'add_nh',
+                "dst_ip": dst_ip,
                 "config_file": FG_ECMP_CFG,
                 "exp_flow_count": exp_flow_count,
                 "add_nh_port": withdraw_nh_port},
             qlen=1000,
             log_file=log_file)
 
 
-    withdraw_nh_bank = bank_0_port
-    for nexthop, port in ip_to_port.items():
-        if port in withdraw_nh_bank:
-            duthost.shell("vtysh -c 'configure terminal' -c 'no {} {} {}'".format(ipcmd, prefix, nexthop))
+    ### Simulate route and link flap conditions by toggling the route
+    ### and ensure that there is no orch crash and data plane impact
+    logger.info("Simulate route and link flap conditions by toggling the route "
+                "and ensure that there is no orch crash and data plane impact")
+    nexthop_to_toggle = ip_to_port.keys()[0]
 
+    cmd = "for i in {1..50}; do "
+    cmd = cmd + vtysh_base_cmd
+    cmd = cmd + "  -c 'no {} {} {}';".format(ipcmd, prefix, nexthop_to_toggle)
+    cmd = cmd + " sleep 0.5;"
+    cmd = cmd + vtysh_base_cmd
+    cmd = cmd + "  -c '{} {} {}';".format(ipcmd, prefix, nexthop_to_toggle)
+    cmd = cmd + " sleep 0.5;"
+    cmd = cmd + " done;"
 
-    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.withdraw_bank.log".format(test_time)
+    configure_dut(duthost, cmd)
+    time.sleep(30)
+
+    result = duthost.shell(argv=["pgrep", "orchagent"])
+    pytest_assert(int(result["stdout"]) > 0, "Orchagent is not running")
+
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.hash_check_post_flap_test.log".format(test_time)
+    ptf_runner(ptfhost,
+            "ptftests",
+            "fg_ecmp_test.FgEcmpTest",
+            platform_dir="ptftests",
+            params={"test_case": 'bank_check',
+                "dst_ip": dst_ip,
+                "exp_flow_count": exp_flow_count,
+                "config_file": FG_ECMP_CFG},
+            qlen=1000,
+            log_file=log_file)
+
+
+    ### Send the same flows again, but disable all next-hops in a bank
+    ### to test flow redistribution to the other bank
+    logger.info("Send the same flows again, but disable all next-hops in a bank "
+                "to test flow redistribution to the other bank")
 
-    time.sleep(1)
+    withdraw_nh_bank = bank_0_port
+
+    cmd = vtysh_base_cmd
+    for nexthop, port in ip_to_port.items():
+        if port in withdraw_nh_bank:
+            cmd = cmd + " -c 'no {} {} {}'".format(ipcmd, prefix, nexthop)
+    configure_dut(duthost, cmd)
+    time.sleep(3)
 
     exp_flow_count = {}
     flows_per_nh = NUM_FLOWS/len(bank_1_port)
     for port in bank_1_port:
         exp_flow_count[port] = flows_per_nh
 
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.withdraw_bank.log".format(test_time)
+
     ptf_runner(ptfhost,
```

Comment:
can we optimize `ptf_runner` this where we make it as fixture and pass only information needed for given case. Most of arguments seems fixed/constant in each case. 

Eg: everflow/test_everflow_testbed.py to check

Reply:
Can this improvement be taken as part of next iteration of the test? I would prefer having this PR checked in due to the length of time it has been open. The next PR will be opened shortly after this by Kavin.

Reply:
@anish-n can we merge next change also as part of this PR and optimize ? Wondering what is blocked on merge on this PR ?

Reply:
There is a different author for that PR, he already has an open PR for minigraph parsing enhancement which needs to go in after this https://github.com/Azure/sonic-mgmt/pull/2500, further, warm boot test case will also be added by Kavin and included in follow up PR.

Reply:
I think these are small enhancements. so better we can do now and then do not need follow up.

Reply:
Makes sense, have optimized PTF runner and now perform this part via a helper function, thanks for the comment!